src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,284 @@
+        } catch (IOException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    static void addDeadline(String userInput) throws DukeException {
+        String deadline = userInput.substring((8));
+        deadline = deadline.replace(" ", "");
+
+        if (deadline.isEmpty()) {
+            throw new DukeException(MISSING_DEADLINE_CONTENT_MESSAGE);
+        } else {
+            Task newTask = new Deadline(userInput.substring(9, userInput.indexOf("/")),
+                    userInput.substring(userInput.indexOf("/by") + 4));
+            addTask(newTask);
+        }
+        try {
+            writeToFile("duke.txt", tasks);
+        } catch (IOException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    static void addEvent(String userInput) throws DukeException {
+        String event = userInput.substring((5));
+        event = event.replace(" ", "");
+
+        if (event.isEmpty()) {
+            throw new DukeException(MISSING_EVENT_CONTENT_MESSAGE);
+        } else {
+            Task newTask = new Event(userInput.substring(6, userInput.indexOf("/")),
+                    userInput.substring(userInput.indexOf("/at") + 4));
+            addTask(newTask);
+        }
+        try {
+            writeToFile("duke.txt", tasks);
+        } catch (IOException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    static void setTaskToBeDone(String userInput) {
+        int selectedTask = Integer.parseInt(userInput.substring(5)) - 1;
+        Task thisTask = tasks.get(selectedTask);
+        thisTask.setAsDone();
+        tasks.set(selectedTask, thisTask);
+        printSeparationLine();
+        System.out.println(MARK_AS_DONE_MESSAGE);
+        System.out.println(LONG_SPACE + "[" + thisTask.getStatusIcon() + "] " +
+                thisTask.getDescription());
+        printSeparationLine();
+    }
+
+    static void deleteTask(String userInput) {
+        int selectedTask = Integer.parseInt(userInput.substring(7)) - 1;
+        Task thisTask = tasks.get(selectedTask);
+        System.out.println(REMOVE_TASK_MESSAGE);
+
+        tasks.remove(selectedTask);
+        numberOfTasks--;
+        addTask = false;
+        echo(thisTask, addTask);
+    }
+
+    static void displayList() {
+        printSeparationLine();
+        switch (numberOfTasks) {
+        case 0:
+            System.out.println(EMPTY_LIST_MESSAGE);
+            break;
+        case 1:
+            System.out.println(ONE_TASK_IN_LIST_MESSAGE);
+            break;
+        default:
+            System.out.println(MORE_THAN_ONE_TASK_IN_LIST_MESSAGE);
+        }
+
+        for (int i = 0; i < numberOfTasks; i++) {
+            System.out.println(SHORT_SPACE + (i + 1) + "." + tasks.get(i));
+        }
+        printSeparationLine();
+        System.out.println();
+    }
+
+    static void greet() {
+        System.out.println(GREET_MESSAGE);
+    }
+
+    static void printSeparationLine() {
```

Comment:
good use to create helper print functions here

Reply:
Thank you for your compliment! :)